hpx/util/get_and_reset_value.hpp

```diff
@@ -31,9 +31,9 @@ namespace hpx { namespace util
     template <typename T>
     inline T get_and_reset_value(std::atomic<T>& value, bool reset)
     {
-        T result = value.load();
+        T result = value.load(std::memory_order_acquire);
         if (reset)
-            value.store(0);
+            value.store(0, std::memory_order_release);
```

Comment:
would it make sense to replace the code with something like this:
```
if (reset)
    return value.exchange(0, std:.memory_order_acq_rel);
else
    return value.load(std::memory_order_relaxed);
```
?

Reply:
I'm not sure, if using `value.load(std::memory_order_relaxed);` would be sufficient, but otherwise this could be done, yes.

Reply:
I'm not sure about using relaxed either. The idea was: when reading the value, we don't require any ordering since it shouldn't matter if we read an old value.

Reply:
Makes sense. Will fix.